src/components/PadBtn/PadBtn.tsx

```diff
@@ -13,34 +19,52 @@ const PadBtn: React.FC<Props> = ({ input, calc, setCalc }) => {
         nums: [""],
         result: 0,
       });
+      setPastEquations([]);
       return;
     }
 
     if (input === "=") {
+      if (newCalc.result === 0)
+        newCalc.result = parseFloat(newCalc.displayedString);
+      setPastEquations(
+        pastEquations.concat(
+          <div key={pastEquations.length} className="history">
+            <p>{newCalc.displayedString}</p>
+            <p>= {newCalc.result}</p>
+          </div>
+        )
+      );
+      if (newCalc.result !== 0)
+        newCalc.displayedString = newCalc.result.toString();
+      newCalc.result = 0;
+      setCalc(newCalc);
       return;
     }
 
     if (input === "<-") {
       newCalc.displayedString = newCalc.displayedString.slice(0, -1);
+      if (newCalc.displayedString === "") newCalc.displayedString = "0";
       getCalculation(newCalc);
       setCalc(newCalc);
       return;
     }
 
-    if (newCalc.displayedString.charAt(0) === "0")
+    if (input === ",") input = ".";
+
+    if (newCalc.displayedString.charAt(0) === "0" && isNumber(input))
       newCalc.displayedString = newCalc.displayedString.slice(0, -1); //clear first zero
 
-      let isLessThan8 = newCalc.nums[newCalc.nums.length - 1].length < 8;
-      if (isLessThan8) newCalc.displayedString += input; //need improvement
+    let isLessThan8 = newCalc.nums[newCalc.nums.length - 1].length < 8;
+    if (isLessThan8) newCalc.displayedString += input; //need improvement
 
     getCalculation(newCalc);
     console.log(newCalc);
     setCalc(newCalc);
   };
 
-  // const isNumber = (str: string) => {
-  //   return !isNaN(parseFloat(str));
-  // };
```

Comment:
Is this commented code necessary?  It looks like you are using the same logic already on the next lines.

Reply:
Even tho I turn the string into two separated arrays. One with numbers and the other one with equation signals. I want to check if the input is a number before adding it to the string, and it is a special tool I will use in future commits